libc/stdio_scanf.c

```diff
@@ -0,0 +1,3701 @@
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_white_signs)
+{
+
+	char tested_str[50];
+	char chr[50];
+	char spc[] = "stdio fscanf test/0x0005";
+
+	sprintf(tested_str, "%s", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+}
+
+TEST(stdio_scanf_str_chr_ptr, pointers)
+{
+
+	char tested_str[50];
+	void **chr;
+	char spc[] = "0x12ab34cd";
+	sprintf(tested_str, "%p", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(spc, chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(spc, chr);
+
+	unlink(std_path);
+	memset(tested_str, 0, sizeof(tested_str));
+	rewind(rd);
+
+	sprintf(tested_str, "%p", (void *)std_path);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(std_path, chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(std_path, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(std_path, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%p", &chr));
+	TEST_ASSERT_EQUAL_PTR(std_path, chr);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_chain)
+{
+
+	char tested_str[1500];
+	char chr[50];
+	char spc[1500];
+	char ret[1500];
+	strcpy(spc, "\\0");
+	for (int i = 1; i < 255; i++) {
+		strcat(spc, "\\");
+		sprintf(chr, "%d", i);
+		strcat(spc, chr);
+		memset(chr, 0, sizeof(chr));
+	}
+	// printf("\n%s\n",buff);
+	sprintf(tested_str, "%s", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%s", ret));
+	TEST_ASSERT_EQUAL_STRING(spc, ret);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%s", ret));
+	TEST_ASSERT_EQUAL_STRING(spc, ret);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%s", ret));
+	TEST_ASSERT_EQUAL_STRING(spc, ret);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%s", ret));
+	TEST_ASSERT_EQUAL_STRING(spc, ret);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_overflow)
+{
+
+	char tested_str[1500];
+	char chr[10];
+	char buff[] = "Something is writen here with more chars then chr";
```

Comment:
Few remarks about this test as a whole:
 - this test tests undefined behavior as defined in posix:
   > If this object does not have an appropriate type, or if the result of the conversion cannot be represented in the space provided, the behavior is undefined.
 - the `tested_str` is way too big to hold a one sentence 
 - I think this test is not memory safe as it may attempt to write outside of buffer